lib/logstash/filters/geoip.rb

```diff
@@ -235,4 +237,20 @@ def tag_unsuccessful_lookup(event)
     @tag_on_failure.each{|tag| event.tag(tag)}
   end
 
+  def apply_geodata(geo_data_hash, event)
+    # don't do anything more if the lookup result is nil?
```

Comment:
since we check on line 174 if geo_data_hash is empty, should we do that check first and then call populate_geo_data being sure it is not empty/nil?